### ISSUE TITLE
Button: Space should only activate on keyup

### ISSUE THREAD

**Issue by sh0ji** [w3c/aria-practices#610]
The synthetic click event that happens on native `<button>` elements occurs on keydown for Enter and keyup for Space, but the button pattern doesn't describe this behavior and the example does not exhibit it. For instance, holding down the space bar on the "Mute" button toggles it off and on until you release the space bar--it should only trigger once on key release.

I tested this behavior using [this codepen example](https://codepen.io/sh0ji/pen/vdXraZ) on Firefox, Chrome, and Safari on Mac to be certain about it. Holding down space bar on the "real button" only triggers the click even on release, whereas holding down enter triggers it repeatedly until release. I added a "fake button" to demonstrate how this behavior can be replicated in JavaScript.

**Comment by kleinfreund** [comment]
I will address this issue in a follow-up pull request to #609, once it is merged. Have it implemented locally.

**Comment by mcking65** [comment]
@kleinfreund, are you still planning to submit a PR for this?

@spectranaut, does this impact button tests in PR #849?

**Comment by kleinfreund** [comment]
@mcking65 I didn’t have this in mind anymore. I simply forgot about it. Sorry for that; I’m on it now.

One thing: For the toggle button’s click event, it is explicitly checked whether the target element is something like a button (i.e. either *is* a `<button>` tag or *has* the `button` role).

```js
if (event.target.getAttribute('role') === 'button' || event.target.tagName === 'button') {
  toggleButtonState(event);
}
```

This check is not in place for any other event and only for the toggle button. In particular, the action button (implemented as a `<div>` element with role `button`) just activates the button without this check.

I would like to remove this check. Currently, it’s not helpful because the examples will fulfill either criteria anyway. Otherwise, we would need to introduce such safety checks in a lot of other places, too.

**Comment by spectranaut** [comment]
hi @mcking65, nope, this bug does not effect the regression tests. The tests send a key and wait for a state change -- they do not test whether the state change happens at the time of keypress or keyup.

**Comment by kleinfreund** [comment]
This can be closed, no?

**Comment by sh0ji** [comment]
Yep, this was resolved by https://github.com/w3c/aria-practices/pull/858. Closing.
